update user info

segunolalive opened the pull request.

segunolalive committed:
make webpack run with local server

segunolalive committed:
Merge pull request #24 from segunolalive/webpack-devserver

chore- Webpack devserver

segunolalive committed:
Add bundle.js script to index.html

segunolalive committed:
change login/signup/logout alert location

segunolalive committed:
reimplement logout component

segunolalive committed:
Refactor login

segunolalive committed:
redirect unauthenticated guests from dashboard

segunolalive committed:
reimplement sign up component and action

segunolalive committed:
Merge pull request #25 from segunolalive/client-auth

Client auth

segunolalive committed:
get user info on login/signup

segunolalive committed:
Merge pull request #26 from segunolalive/feature/add-user-info

get user info on login/signup

segunolalive committed:
create action creator for getting borrowed books

segunolalive committed:
add authentication header to axios requests

segunolalive committed:
make dashboard fetch borrowed books

segunolalive committed:
send user data on sign in

segunolalive committed:
set on authorization token on client login

segunolalive committed:
add book title to message on returning book

segunolalive committed:
allow user return books

segunolalive committed:
Merge pull request #27 from segunolalive/client-fetch-borrowed

Client fetch borrowed

segunolalive committed:
reduce toast duration to 2.5 secs

segunolalive committed:
implement borrowing and returning books on client

segunolalive committed:
BUGFIX: update book quantity when returned

segunolalive committed:
Add library actions and action creators

segunolalive committed:
Merge pull request #29 from segunolalive/feature/library-component

Feature/library component

segunolalive committed:
save redux store to disk

segunolalive committed:
Add and fetch book categories

segunolalive committed:
edit flash messages

segunolalive committed:
add admin page components

segunolalive committed:
add book detail page

segunolalive committed:
mop up session storage on logout

segunolalive committed:
fetch and display book categories

segunolalive committed:
prevent logged in users from visting home page

segunolalive committed:
STYLE: Darken all toast notifications

segunolalive committed:
correct edit book test assertion

segunolalive committed:
Merge pull request #30 from segunolalive/feature/library-component

Feature/library component

segunolalive committed:
refactor header component

segunolalive committed:
Adjust side bar width

segunolalive committed:
Add updateUserInfo route

segunolalive committed:
add beforeUpdate model hook to user model

segunolalive committed:
Merge pull request #32 from segunolalive/feature/update-user

Feature/update user

vynessa commented on .eslintrc:
```diff
@@ -0,0 +1,41 @@
+            "MethodDefinition": true,
+            "ClassDeclaration": true
+        }
+    }]
```
After this line use the max length property 
`"max-len": [1, 80, 2]`. You will find this handy as this is the organization's style guide.

segunolalive replied:
Thank you @vynessa. What line are you referring to?

vynessa commented on README.md:
```diff
@@ -0,0 +1,63 @@
+[![Build Status](https://travis-ci.org/segunolalive/helloBooks.svg?branch=master)](https://travis-ci.org/segunolalive/helloBooks)
```
Good detailing in your readme file. It can be better and more comprehensive. Take a look at [this](https://github.com/vynessa/dman/blob/develop/README.md). 

Is documenting your API using any documentation  generator one of the requirements for this project? If no, i'd advice you look up [slate](https://github.com/lord/slate) or swagger for this.

vynessa commented on client/actions/bookActions.js:
```diff
@@ -0,0 +1,70 @@
+import actionTypes from './actionTypes';
```
I can see some es5 syntax in this file e.g using the function keyword. Make sure to correct accordingly.

segunolalive replied:
I'm going to remove the contents, I'm actually not using that file anymore. Thanks

vynessa commented on client/components/Admin.js:
```diff
@@ -0,0 +1,97 @@
+import Header from './header/Header';
+import BookForm from './BookForm';
+
+// import jQuerySelect from '../static/jquery/select';
```
Remove any form of zombie code.

vynessa commented on client/components/auth/Login.js:
```diff
@@ -0,0 +1,136 @@
+                          </div>
+                          <div>
+                            <p className="center">Don't have an account?
+                              <Link to="/signup"> Sign up</Link>
```
Investigate this file to see what is behind the red line display.

segunolalive replied:
It's because I didn't use html escape codes for the apostrophe

vynessa commented on client/components/auth/SignUp.js:
```diff
@@ -0,0 +1,184 @@
+    .then(
+      (response) => {
+        Materialize.toast(
+          'Welcome To Hello Books. This is you dashboard',
```
Fix the spelling error here.

segunolalive replied:
Okay, Thanks

vynessa commented on client/components/dashboard/Dashboard.js:
```diff
@@ -0,0 +1,86 @@
+    this.props.returnBook(this.props.user.id, bookId);
+  }
+
+  readBook (id) {
```
Make sure to remove unnecessary functions or block of codes.

vynessa commented on client/components/library/BookDetail.js:
```diff
@@ -0,0 +1,121 @@
+  book: bookReducer.currentBook,
+});
+
+// const mapDispatchToProps = dispatch => bindActionCreators({ borrowBook }, dispatch);
```
Zombie code.

vynessa commented on client/index.js:
```diff
@@ -0,0 +1,20 @@
+import store from './store/store';
+import App from './components/App';
+import './static/index.scss';
+// import registerServiceWorker from './registerServiceWorker';
```
Zombie code.

vynessa commented on client/index.js:
```diff
@@ -0,0 +1,20 @@
+  </Provider>,
+  document.getElementById('root')
+);
+// registerServiceWorker();
```
Be careful not to merge zombie code (commented out code) into your production branch.

vynessa commented on client/static/index.scss:
```diff
@@ -0,0 +1,157 @@
+$coolBlue: #032442;
```
Just as the hound bot has warned, your variable names in sass/scss should be in this format `$cool-blue` or `$dark-red`.

segunolalive replied:
okay. Noted. Thank you

vynessa commented on client/sw.js:
```diff
@@ -0,0 +1,48 @@
+// In production, we register a service worker to serve assets from local cache.
```
For multiple lines of comment, it is much preferable to use this `/** */`.

segunolalive replied:
Thank you. 👍

vynessa commented on client/sw.js:
```diff
@@ -0,0 +1,48 @@
+          installingWorker.onstatechange = () => {
+            if (installingWorker.state === 'installed') {
+              if (navigator.serviceWorker.controller) {
+                // At this point, the old content will have been purged and
```
Same should be applied here.

vynessa commented on server/config/config.js:
```diff
@@ -0,0 +1,19 @@
+    host: 'localhost',
+    dialect: 'postgres',
+  },
+  test: {
```
Great hiding your database details.

vynessa commented on server/controllers/books.js:
```diff
@@ -0,0 +1,346 @@
+import { Book, BorrowedBook, BookCategory } from '../models';
```
Check that the return tag in your JSDocs for your methods state clearly the right data being returned.

vynessa commented on server/helpers/helpers.js:
```diff
@@ -0,0 +1,35 @@
+  return user;
+};
+
+/*
```
Zombie code. Kill it!

segunolalive replied:
I didn't quite get that

segunolalive replied:
Okay. Done! Thank you

vynessa commented on server/migrations/20170821170950-book category.js:
```diff
@@ -0,0 +1,28 @@
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
```
You can use the DATEONLY method for your createdAt and updatedAt props if you have no use for the entire value DATE returns.

vynessa commented on server/routes/index.js:
```diff
@@ -0,0 +1,25 @@
+
+
+export default (app) => {
+  app.post('/api/v1/users/signup', userController.createUser);
```
You should take a look at `router.route` and consider using it for routing.

vynessa commented on server/test/controllers/book.spec.js:
```diff
@@ -0,0 +1,192 @@
+        .get('/api/v1/books')
+        .expect(200)
+        .end((err, res) => {
+          assert.equal(res.status, 200);
```
For your tests, you should not only test your status codes. To make this more comprehensive, test the response payload also.

segunolalive replied:
Noted. Thank you

vynessa commented on templates/index.html:
```diff
@@ -0,0 +1,54 @@
+        <title>Hello Books</title>
+    </head>
+    <body>
+        <header>
```
You can abstract the header or navbar and make it a component.

segunolalive replied:
It is. This is the mock folder

vynessa commented on webpack.prod.config.js:
```diff
@@ -0,0 +1,72 @@
+      }
+    })
+  ],
+};
```
Good job so far.

segunolalive committed:
Implement admin notifications

segunolalive committed:
STYLE: limit line length to 80 characters

segunolalive committed:
REFACTOR: extract admin status check to dedicated middleware function

segunolalive committed:
implemet borrowing limit

segunolalive committed:
FEATURE: Implement admin notification

segunolalive committed:
change implementation of book categories

segunolalive committed:
Miscellaneous: Clean up

segunolalive committed:
BUGFIX: correct JWT arguement order

segunolalive committed:
remove unused book actions file

segunolalive committed:
STYLE: refactor scss style to align with conventions

segunolalive committed:
feature(borrowing): Set borowing limits

segunolalive committed:
Merge pull request #32 from segunolalive/feature/update-user

Feature/update user

segunolalive committed:
Merge branch 'feature/limit_borrowing' of https://github.com/segunolalive/helloBooks into feature/limit_borrowing

segunolalive committed:
Merge pull request #32 from segunolalive/feature/update-user

Feature/update user

segunolalive committed:
Merge branch 'feature/limit_borrowing' of https://github.com/segunolalive/helloBooks into feature/limit_borrowing

segunolalive committed:
Merge pull request #33 from segunolalive/feature/limit_borrowing

Feature/limit borrowing

segunolalive committed:
feature(history): add user borrwing history

segunolalive committed:
Merge pull request #34 from segunolalive/feature/borrowing-history

feature(history): add user borrwing history

segunolalive committed:
feature(admin-notification): notify admin of every transaction

segunolalive committed:
Merge pull request #35 from segunolalive/feature/admin-notification

feature(admin-notification): notify admin of every transaction

segunolalive committed:
feature(book detail): reimplement book detail logic

segunolalive committed:
Merge pull request #36 from segunolalive/feature/view-book

feature(book detail): reimplement book detail logic

segunolalive committed:
chore(materialize): clean up materialize references

segunolalive committed:
Merge pull request #37 from segunolalive/chore/materialize-toast

chore(materialize): clean up materialize references

segunolalive committed:
chore(conflict): resolve conflicts

tduyile04 commented on client/components/dashboard/Borrowed.js:
```diff
@@ -0,0 +1,63 @@
+          <div className="card-stacked">
+            <div className="card-content">
+              <h6 className="bold-text black-text">
+                {book.title.toUpperCase() || null}
```
Nitpick: why is there a value defaulting to null if the title property doesn't exist.

tduyile04 commented on client/components/library/BookDetail.js:
```diff
@@ -0,0 +1,138 @@
+    );
+    const redirectToEditPage = this.state.editRedirect;
+    const redirectToLibrary = this.state.deleteRedirect;
+      return(
```
Nitpick: Your indentation here is off by 2 spaces

tduyile04 commented on client/components/library/BooksTable.js:
```diff
@@ -0,0 +1,49 @@
+        </tr>
+      </thead>
+      <tbody>
+        {books}
```
Your books defaults to null if the bookList props passes a 'Falsy' value down to the BooksTable component. Don't you think a feedback message would be better for your user experience ??

tduyile04 commented on client/components/library/Categories.js:
```diff
@@ -0,0 +1,41 @@
+        onChange={event => props.onChange(event)}
+      >
+        <option value='' disabled>{props.text}</option>
+        {options}
```
Check previous comment!!!

segunolalive committed:
add app.json for heroku pipeline

segunolalive committed:
Merge pull request #38 from segunolalive/chore/heroku-pipeline

add app.json for heroku pipeline

segunolalive committed:
fix: refactor admin component logic

segunolalive committed:
Merge pull request #39 from segunolalive/hotfix/admin-book

fix: refactor admin component logic

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
feature(profile): add profile update component

segunolalive committed:
add profile update component

segunolalive committed:
Merge branch 'feature/update-profile' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
bug(google): temporarily disable react-google-login

segunolalive committed:
bug(google): temporarily disable react-google-login

segunolalive committed:
Merge branch 'feature/update-profile' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
Merge branch 'feature/update-profile' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
Merge branch 'feature/update-profile' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
Merge branch 'feature/update-profile' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
Merge branch 'feature/update-profile' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
feature(set-up): initial set up for node mailer

segunolalive committed:
feature(forgot-password): add forgot password form

segunolalive committed:
feature(pagination): set up metadata pagination format

segunolalive committed:
chore(jsdoc): make book controller expressly return http response object

segunolalive committed:
refactor(notifications): refactor and comment notifications controller

segunolalive committed:
chore(jsdoc): comment history component

segunolalive committed:
chore(jsdoc): add jsdocs for Header component

segunolalive committed:
chore(jsdoc): add jsdocs for dashboard component

segunolalive committed:
chore(jsdoc): add jsdocs to Home component

segunolalive committed:
chore(config): update webpack config

segunolalive committed:
chore(jsdoc): remove extension from component imports

segunolalive committed:
chore(jsdocs): update getBookCategory docs

segunolalive committed:
chore(jsdocs): add jsdocs for auth components

segunolalive committed:
chore(config):make eslint resolve jsx extensions

segunolalive committed:
chore(jsdocs): add jsdocs to admin component

segunolalive committed:
chore(jsdocs): update docs for authentication components and Header component

segunolalive committed:
chore(extension): change component extensions from .js to .jsx

segunolalive committed:
chore(jsdoc): add documentation for library component

segunolalive committed:
chore(jsdoc): add jsdocs for app entry point

segunolalive committed:
chore(jsdocs): minor changes

segunolalive committed:
chore(jsdocs): addmjsdocs to miscellaneous files in client

segunolalive committed:
chore(localStorage): switch sessionStorage to LocalStorage

segunolalive committed:
feature(style): adjust index.html fallback styles

segunolalive committed:
chore(clean-up): miscellaneous clean up

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into chore/jsdocs

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into chore/jsdocs

segunolalive committed:
Merge pull request #41 from segunolalive/chore/jsdocs

Chore jsdocs

segunolalive committed:
Merge branch 'chore/jsdocs' of https://github.com/segunolalive/helloBooks into chore/jsdocs

segunolalive committed:
chore(prop-types): remove proptype declarations from production build

segunolalive committed:
Merge pull request #42 from segunolalive/chore/jsdocs

Chore/jsdocs

segunolalive committed:
chore(dependencies): add babel plugins for react-google-login

segunolalive committed:
chore(dependencies): add babel plugins for react-google-login

segunolalive committed:
Merge branch 'chore/jsdocs' of https://github.com/segunolalive/helloBooks into chore/jsdocs

segunolalive committed:
Merge branch 'chore/jsdocs' of https://github.com/segunolalive/helloBooks into chore/jsdocs

segunolalive committed:
Merge branch 'chore/jsdocs' of https://github.com/segunolalive/helloBooks into chore/jsdocs

segunolalive committed:
Merge pull request #43 from segunolalive/chore/jsdocs

chore(dependencies): add babel plugins for react-google-login

segunolalive committed:
feature(notification): save user activities to notifications table

segunolalive committed:
chore(transactions): rename transactons to transactionController

segunolalive committed:
Merge pull request #44 from segunolalive/feature/admin-notification

#149760479  admin notification

segunolalive committed:
feature(error): explicitly handle API request errors in action creators on client

segunolalive committed:
feature(auth): abstract error handling in aignup and login components to action creators

segunolalive committed:
feature(jwt): handle token expiration

segunolalive committed:
chore(response statements): remove unnecessary success key from http response body

segunolalive committed:
chore(user-controller): explicitly return response objects in user contoller

segunolalive committed:
Bug(hotfix): fix admin notification.

Context: Notifications displayed were only for admin user

segunolalive committed:
fix(authentication): change status code from 400 401 when unauthenticated

segunolalive committed:
fix(status codes): change status code from 201 to 200 when book is returned

segunolalive committed:
fix(status codes): correct satus code if unauthenticated

correct status code when returning book

segunolalive committed:
resolve conflict

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/mail-users

Update feature branch with upstream development branch

segunolalive committed:
Merge pull request #46 from segunolalive/error-handling

Error handling

segunolalive committed:
Merge pull request #45 from segunolalive/hotfix/admin-notifications

fix admin notification.

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/mail-users

segunolalive committed:
feature(defaulter emails): mail users who do not returned as when due

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/update-profile

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/forgot-password

segunolalive committed:
bug(authorization-header): fix authentication problem using the Authorization header

segunolalive committed:
bug(authentication): fix issue with Authorization request header

segunolalive committed:
chore(rename): change update Profile component file extension to jsx

segunolalive committed:
chore(service worker): transpile service worker file

segunolalive committed:
chore(authentication): rename authentication middleware

segunolalive committed:
feature(history): edit history table appearance

segunolalive committed:
feature(update-profile): add update profile function
ality

segunolalive committed:
feature(update-profile): update profile info and password

segunolalive committed:
Merge pull request #40 from segunolalive/feature/update-profile

#149760579 Update profile

segunolalive committed:
feature(notifications): add time reference to nofication messages

bugfix: create notifications on initial borrowing

segunolalive committed:
feature(google-auth): signup unregistered users on google login

segunolalive committed:
Merge pull request #47 from segunolalive/hotfix/admin-notifications

feature(notifications): add time reference to notification messages

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/forgot-password

segunolalive committed:
feature(cron): ensure process exits after job runs

segunolalive committed:
chore(dependencies): update sequelize

segunolalive committed:
feature(forgpt-password): add forgot password component and action creator

segunolalive committed:
feature(reset-password): add reset password mail controller

segunolalive committed:
chore(jsdoc): add jsdoc for auth component constructor

segunolalive committed:
chore(clean-up): remove unused arguements in actions

segunolalive committed:
Merge branch 'feature/mail-users' of https://github.com/segunolalive/helloBooks into feature/forgot-password

segunolalive committed:
feature(forgot-password): rename forgot-password files

segunolalive committed:
feature(forgot-password): send email to user

segunolalive committed:
feature(forgot-password): display loader when mailing user

segunolalive committed:
feature(forgot-password): remove api version from route declarations

segunolalive committed:
feature(forgot-password): refactor jwt helper

segunolalive committed:
feature(forgot-password): update addBook component logic

segunolalive committed:
chore(readme): update readme

segunolalive committed:
feature(forgot-password): change handleSubmit prop to onSubmit on forgot password form

segunolalive committed:
feature(forgot-password): add password reset form

segunolalive committed:
feature(forgot-password): add Password container component

- Contain business logic for forgot password and password reset in a
single container

segunolalive committed:
feature(forgot-password): change url in reset password mail

- Set token in search param

segunolalive committed:
feature(forgot-password): add thunk for resetting password.

segunolalive committed:
chore(readme): add link for api docs

segunolalive committed:
chore(forgot-password):  server-side validation on password reset

segunolalive committed:
feature(reset-password): redirect to login page on password update

segunolalive committed:
feature(reset-password): reimplement deletEmptyFields and trimFields

- make both functions pure

segunolalive committed:
feature(reset-password): reimplement deletEmptyFields and trimFields

- make both functions pure

segunolalive committed:
Merge branch 'feature/forgot-password' of https://github.com/segunolalive/helloBooks into feature/forgot-password

segunolalive committed:
feature(forgot-password): update schema

- add authId and token fields to user schema for social signup and
password reset respectively
- Update book and book categoty schema with category column

segunolalive committed:
feature(forgot-password): Add License

- MIT License

segunolalive committed:
feature(forgot-password): abstract error notifications

segunolalive committed:
feature(forgot-password): file name changes

segunolalive committed:
feature(forgot-password): add authentictation utilities

- completes forgot password
- completes google login

segunolalive committed:
feature(forgot-password): move db config parameters to environment

segunolalive committed:
feature(forgot-password): change response format

segunolalive committed:
feature(forgot-password): add tests for helper functions

segunolalive committed:
feature(forgot-password): limit resetPasswordToken use to once

segunolalive committed:
Merge pull request #48 from segunolalive/feature/forgot-password

#149760771 Users should  be  able  to  reset  their  passwords

segunolalive committed:
feature(forgot-password): ladd google client Id

segunolalive committed:
bug(unresolved-extensions): move files

segunolalive committed:
Merge pull request #49 from segunolalive/bug/unresolved-extensions

fix unresolved extensions

segunolalive committed:
feature(pagination): refactor transaction controllers

segunolalive committed:
feature(pagination): add server-side pagination for books

segunolalive committed:
Merge branch 'development' of https://github.com/segunolalive/helloBooks into feature/google-signin

segunolalive committed:
Merge branch 'development' into feature/pagination

segunolalive committed:
Merge pull request #50 from segunolalive/feature/google-signin

Feature/google signin

segunolalive committed:
feature(pagination): add client-side pagination

segunolalive committed:
feature(pagination): add client-side pagination for books

segunolalive committed:
feature(pagination): add book search

- change library appearance

segunolalive committed:
feature(pagination): extract queryFromObject to separate file

segunolalive committed:
feature(pagination): add pagination to admin notifications

segunolalive committed:
feature(pagination): add removal book filter

segunolalive committed:
feature(pagination): make header fixed

segunolalive committed:
feature(pagination): move loader component to utils

segunolalive committed:
feature(pagination): extract client-side pagination utilities

segunolalive committed:
feature(pagination): remove redundant code from book controller

segunolalive committed:
feature(pagination): add server-side pagination to admin notifications

segunolalive committed:
feature(pagination): update test config

segunolalive committed:
Merge branch 'feature/pagination' of https://github.com/segunolalive/helloBooks into feature/pagination

segunolalive committed:
feature(pagination): add client-side pagination

segunolalive committed:
feature(pagination): create transaction reducer

segunolalive committed:
feature(pagination): set up client-side testing

segunolalive committed:
feature(pagination): add inline form validation

segunolalive committed:
feature(pagination): update pagintion logic

segunolalive committed:
feature(pagination): add file upload

segunolalive committed:
feature(pagination): add file upload

segunolalive committed:
Merge branch 'feature/pagination' of https://github.com/segunolalive/helloBooks into feature/pagination

segunolalive committed:
Merge pull request #51 from segunolalive/feature/pagination

pagination

segunolalive committed:
ad stage-2 babel presets

saqibameen committed:
UPDATE: Add Space between path value and component

segunolalive committed:
Merge pull request #52 from MrSaqibAmeen/patch-1

UPDATE: Add Space between path value and component

segunolalive committed:
chore(tests): add client-side tests

segunolalive committed:
chore(tests): add api documentation

segunolalive committed:
chore(tests): add e2e tests

segunolalive committed:
chore(tests): update jsdocs

segunolalive committed:
Merge branch 'chore/tests' of https://github.com/segunolalive/helloBooks into chore/tests

segunolalive committed:
Merge pull request #53 from segunolalive/chore/tests

Chore/tests

segunolalive committed:
feature(delete-book): add delete book confirmation

segunolalive committed:
feature(delete-book): refactor utitlity functions

segunolalive committed:
Merge pull request #54 from segunolalive/feature/delete-book

#153539763 Prevent users from mistakingly deleting books

segunolalive committed:
feature(read-book): refactor file upload

segunolalive committed:
feature(read-book): add more test cases

segunolalive committed:
feature(read-book): make newly added category show up in list

segunolalive committed:
feature(read-book): add book suggestions

segunolalive committed:
feature(read-book): fix navigation links in mobile view

segunolalive committed:
feature(read-book): update tests

segunolalive committed:
feature(read-book): extract error notification into a module

segunolalive committed:
feature(read-book): add ability to read book

segunolalive committed:
feature(read-book): make app offline-first

segunolalive committed:
feature(read-book): format jsDocs

segunolalive committed:
feature(read-book): compress static files in production

segunolalive committed:
feature(read-book): add more validations

segunolalive committed:
feature(read-book): use class properties

segunolalive committed:
feature(read-book): update api documentation

segunolalive committed:
feature(read-book): update styles

- add styles for pdf viewer
- adjust notification section height

segunolalive committed:
feature(read-book): change controller filenames to PascalCase

segunolalive committed:
feature(read-book): rename singletons using PascalCase

segunolalive committed:
feature(read-book): rename bare object export using PascalCase

segunolalive committed:
Merge branch 'feature/read-book' of https://github.com/segunolalive/helloBooks into feature/read-book

segunolalive committed:
Merge branch 'feature/read-book' of https://github.com/segunolalive/helloBooks into feature/read-book

segunolalive committed:
Merge pull request #55 from segunolalive/feature/read-book

#149760707 Users should be able to read books within the app

segunolalive committed:
Revert "#149760707 Users should be able to read books within the app"

segunolalive committed:
Merge pull request #56 from segunolalive/revert-55-feature/read-book

Revert "#149760707 Users should be able to read books within the app"

segunolalive committed:
change order of tests
